include/ofi_mem.h

```diff
@@ -474,11 +478,21 @@ static inline size_t ofi_buf_index(void *buf)
 static inline void *ofi_bufpool_get_ibuf(struct ofi_bufpool *pool, size_t index)
 {
 	void *buf;
+	size_t region_index = index / pool->attr.chunk_cnt;
+
+#ifndef NDEBUG
+	if (region_index >= pool->region_cnt)
+		return NULL;
+#endif
 
-	buf = pool->region_table[(size_t)(index / pool->attr.chunk_cnt)]->
-		mem_region + (index % pool->attr.chunk_cnt) * pool->entry_size;
+	buf = pool->region_table[region_index]->mem_region +
+		(index % pool->attr.chunk_cnt) * pool->entry_size;
+
+#ifndef NDEBUG
+	if (!ofi_buf_hdr(buf)->allocated)
+		return NULL;
+#endif
```

Comment:
Why not just assert on the above 2 checks if they're going to be debug only code?

Reply:
Works for me.